### PR TITLE
Request render on afterRender only as needed

### PR DESCRIPTION
**Pull request by ggetz**
Fixes https://github.com/CesiumGS/cesium/issues/10756

This issue was introduced by https://github.com/CesiumGS/cesium/pull/10607. Basically, if a function is called as part of the `afterRender` step, a new render is always requested. For these primitives, an after render function is required to mark the primitive as `ready` each frame, so it nullifies request render mode.

The fix was to introduce a return parameter to the after render functions to give a bit more control if another frame is needed after. If `true` is returned then another render is requested. If not, then the function is still called, but no additional frames are rendered.

### CODE REVIEW

**Comment by sanjeetsuhag** [comment]
@ggetz Does the change also need to be propagated to `ModelAnimationCollection`? See this [Sandcaslt]e](http://localhost:8080/Apps/Sandcastle/index.html#c=tVdtbyI3EP4rLp8WiRrIy11LSdSE9HpRGzUN3KlSqU7GO8ta8drU9oakp/z3ju1dWF76AaUgJPwynud5xmN74FpZR54ELMGQC6JgSUZgRVnQz2EsmbZ46I+0ckwoMNNWh3ydKkIM/F2CdQ+gUjB3OoUBcaaEjp8r2LMoyiLOTUQBo5ypOVrcqkwo4V6ClVCZvtbPA5IxaeNCCxK4E1rdqlRw5rTZmM1Zqpe2AWRzXcr0SomCuTWB1/YPUxVFUctBAU1hVs7HuV5+MKwAew9mDFyrFDX7NWg+VVmpAjThBtCblyST0sgOyUHMc9eOuiu/oJxwAiw1UOgnuJIyaQc3hPAQ1IW2Iri7qEM6YsZhi6lTmhld3MDcANjELyHk2/7JKe29Pzt71/++E4fOzmjvvHf6vveuGog8fNtD1UA5sFSo+RrnjrmcOv2Aw0zZpH963jRfCMdzNO41xoyWcmsoX2wlxMcIc++XP6B9UuF2osdOcNIE0kZgkNhmDCYGKWXaFJbmWw5/LzHqRqF5FZI6grX6hamlr0HCNryg/+1tYWmafI0LFW75gPid3PQ72EZoMB40O9V04TNiQCqvBD2Kpls0wNzGtL8XzyDH4h8E7Z98t56Nh2LMmcSZkx5+qrnX8Psag1cJcYbxR0h/qvVFoWjx6uVXEV54dhZn//QrK2IOnt2ATFtXwnDDMocntpKn4vHCI1WnetJey2mmfT1G0A+lXfyOWbGQcMMc64Y42G7cUUSpWl+wSedytsLzn3NUSXt1PypcCe7sYX1jtILjUg4QzfYO6/75YaR/NrrEy+Qz5ILLo7GPKBXIZm9HwSHsP2pHcO/INZNSa3Xc4Fcgm73dDTg0be6EfCQTU/LH49L3OAFmu/+mHRg/CqUgJfhM4rF34ZU9pghWx/8LNt/E/JOSwhF8xY/FGF0HjFUD6bpsK1kOvGEY12Ski4UBazHqgcyx+Ae0NdietNlSc4iUXyZ/nDSVHHiCRUaSb6qH+QPKKg3cgIsFGLXlYqGNs9fMCps0q6l2wwUhS6GwKKP4qBnXiIiPySQXlswMlmxYXaYaLFF41VSOSXBMPinxBMYyiQ/6SofXh2TstLV2WIcDQ/G/3EA+drsjb7+JbvFlZor73TjqqVjhbHT2nI7z/yb/V6ilxkylnFmHzwhWTROt5YyZO1BlUhUYfmGr0xpa9yLhsnb3oyjCPmIFlCBJB0gSRdjuDPMaHOXW1ojDbnPpMBVPRKQXe/5bEC6ZtTiTlTKUUNPW5bCL9jtLpQ7V42+YPJK9eLO8f/lrHKSUDrvY3b/SRYFbnv8F) where the model is only animating when I change the camera view.

**Comment by ggetz** [comment]
@sanjeetsuhag I don't think so. Model animations in request render mode require special handling. [This sandcastle shows what you need to do under different circumstances](https://sandcastle.cesium.com/?src=Scene%20Rendering%20Performance.html). To see an animation, you must manually request renders or set the `maximumRenderTimeChange`. The sandcastle you posted does not animate in `main` either for this reasosn.

File: CHANGES.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 - Fixed a bug where the scale of a `Model` was being incorrectly applied to its bounding sphere. [#10855](https://github.com/CesiumGS/cesium/pull/10855)
 - Fixed a bug where rendering a `Model` with image-based lighting while specular environment maps were unsupported caused a crash. [#10859](https://github.com/CesiumGS/cesium/pull/10859)
+- Fixed a bug where request render mode was broken when a ground primitive is added. [#10756](https://github.com/CesiumGS/cesium/issues/10756)
```
Comment (sanjeetsuhag) [inline CHANGES.md:13]: This should be updated with the link to PR.

**Review by sanjeetsuhag — COMMENTED** [review]
Changes look good. I tested a bunch of different scenarios and this looks like the right fix.

File: CHANGES.md (hunk 1 above)
Comment (ggetz) [reply CHANGES.md:13]: We often link to the original issue fixed in `CHANGES.md` - [For instance](https://github.com/CesiumGS/cesium/blob/28ebe20d91b1715b349c5cef92f8a37f4405baf9/CHANGES.md?plain=1#L80)

Comment (sanjeetsuhag) [reply CHANGES.md:13]: Oh, interesting. I never noticed that before, always thought it was only PRs.

**Comment by sanjeetsuhag** [comment]
Thanks @ggetz!

**Pull request merged**